test/cases/sdk_test.go

```diff
@@ -362,6 +362,42 @@ func TestSDKIntegrationRootHandler(t *testing.T) {
 	})
 }
 
+func TestSDKIntegrationWithOverrideHeaderKey(t *testing.T) {
+	sr := tracetest.NewSpanRecorder()
+	provider := sdktrace.NewTracerProvider()
+	provider.RegisterSpanProcessor(sr)
+
+	// Define a function inline that transforms the default header name to a custom header name
+	customHeaderKeyFunc := func() string {
+		return "X-Custom-Trace-ID"
+	}
+
+	router := chi.NewRouter()
+	router.Use(otelchi.Middleware(
+		"foobar",
+		otelchi.WithTracerProvider(provider),
+		otelchi.WithTraceResponseHeaderKey(customHeaderKeyFunc),
+	))
+	router.HandleFunc("/user/{id:[0-9]+}", ok)
+	router.HandleFunc("/book/{title}", ok)
+
+	r0 := httptest.NewRequest("GET", "/user/123", nil)
+	w := httptest.NewRecorder()
+	router.ServeHTTP(w, r0)
+
+	require.Len(t, sr.Ended(), 1)
+	assertSpan(t, sr.Ended()[0],
+		"/user/{id:[0-9]+}",
+		trace.SpanKindServer,
+		attribute.String("http.server_name", "foobar"),
+		attribute.Int("http.status_code", http.StatusOK),
+		attribute.String("http.method", "GET"),
+		attribute.String("http.target", "/user/123"),
+		attribute.String("http.route", "/user/{id:[0-9]+}"),
+	)
+	require.Equal(t, w.Header().Get(customHeaderKeyFunc()), sr.Ended()[0].SpanContext().TraceID().String())
+}
```

Comment:
We should also add a test for the middleware without `otelchi.WithTraceResponseHeaderKey()` option.

We should ensure that the trace id is not added to the default `defaultTraceResponseHeaderKey`.